zerver/lib/test_runner.py

```diff
@@ -225,13 +225,19 @@ def _replacement_destroy_test_db(self: DatabaseCreation,
                        % (self.connection.ops.quote_name(test_database_name),))
 DatabaseCreation._destroy_test_db = _replacement_destroy_test_db
 
-def destroy_test_databases(worker_id: int) -> None:
-    """Modified from the Django original to """
-    database_id = random_id_range_start + worker_id
+def destroy_test_databases(worker_id: Optional[int]=None) -> None:
     for alias in connections:
         connection = connections[alias]
         try:
-            connection.creation.destroy_test_db(number=database_id)
+            # We need this conditional otherwise the call to destroy_test_db gets
+            # confused on which db is being destroyed and tries to append
+            # `number` on to a clone.
+            if worker_id is not None:
+                """Modified from the Django original to """
+                database_id = random_id_range_start + worker_id
+                connection.creation.destroy_test_db(number=database_id)
+            else:
+                connection.creation.destroy_test_db(number=worker_id)
```

Comment:
I think keeping it to be just `connection.creation.destroy_test_db()` is better since `number` isn't a required parameter and defaults to `None`.
Like putting in to different variables might confuse someone.
Also I think we can add a more detailed comment over here explaining how does destruction of the database works in the serial mode and how its different from the parallel mode.
I will be sharing more details of the difference in process and why the function was getting confused and appending database_id twice to the database name over chat.

Reply:
I made this change, updated some comments, and will merge, just because this is a pressing bug.  I'll post comments below for follow-ups.

I definitely want to understand the double `database_id` detail...